### PR TITLE
Use override opts in queries

### PR DESCRIPTION
**Pull request by JaceRockman**

### CODE REVIEW

File: src/clj/fluree/db/query/api.cljc [hunk 1]
```diff
@@ -35,7 +35,8 @@
   [query {:keys [identity did issuer] :as override-opts}]
   (update query :opts (fn [{:keys [max-fuel meta] :as opts}]
                         ;; ensure :max-fuel key is present
-                        (-> (assoc opts :max-fuel max-fuel)
+                        (-> (merge override-opts opts)
```
Comment (dpetran) [inline src/clj/fluree/db/query/api.cljc:38]: Good catch! If we want the override opts to override we'll have to swap the order of the merge:
`(merge opts override-opts)`

File: src/clj/fluree/db/query/api.cljc [hunk 2]
```diff
@@ -35,7 +35,8 @@
   [query {:keys [identity did issuer] :as override-opts}]
   (update query :opts (fn [{:keys [max-fuel meta] :as opts}]
                         ;; ensure :max-fuel key is present
-                        (-> (assoc opts :max-fuel max-fuel)
+                        (-> (merge opts override-opts)
+                            (assoc :max-fuel max-fuel)
```
Comment (dpetran) [inline src/clj/fluree/db/query/api.cljc:39]: Oh, I just realized that the `(assoc :max-fuel max-fuel)` is using the `:max-fuel` from `opts`, so you'll be clobbering the `override-opts`. I think we can simplify the whole thread form:

```
(-> (assoc opts :max-fuel max-fuel)
      (merge opts override-opts)
      ;; get rid of :did, :issuer opts
      (update :identity #(or % (:did opts) (:issuer opts)))
      (dissoc :did :issuer))

Comment (JaceRockman) [reply src/clj/fluree/db/query/api.cljc:39]: Ah, that makes sense, good catch!

**Review by dpetran — APPROVED** [review]
:thread:

**Pull request merged**